core/src/main/scala/cats/syntax/list.scala

```diff
@@ -52,6 +51,33 @@ final class ListOps[A](private val la: List[A]) extends AnyVal {
     toNel.fold(SortedMap.empty[B, NonEmptyList[A]])(_.groupBy(f))
   }
 
+  /**
+   * Groups elements inside this `List` according to the `Order` of the keys
+   * produced by the given mapping monadic function.
+   *
+   * {{{
+   * scala> import cats.data.NonEmptyList
+   * scala> import scala.collection.immutable.SortedMap
+   * scala> import cats.implicits._
+   *
+   * scala> val list = List(12, -2, 3, -5)
+   *
+   * scala> val expectedResult = Option(SortedMap(false -> NonEmptyList.of(-2, -5), true -> NonEmptyList.of(12, 3)))
+   *
+   * scala> list.groupByNelA(num => Option(0).map(num >= _)) === expectedResult
```

Comment:
btw you don't need to do any explicit assertions here, you could just do 
```scala
scala> list.groupByNelA(num => Option(0).map(num >= _))
res0: Option(SortedMap(false -> NonEmptyList(-2, -5), true -> NonEmptyList(12, 3)))
``` 

which will be a little less noise for the reader, while still asserting that the resulting expression is the same as what you typed. :)

Reply:
hmm, this one is pretty interesting
```
[info] > Labels of failing property: 

[info] 'Some(TreeMap(false -> NonEmptyList(-2, -5), true -> NonEmptyList(12, 3)))' is not equal to 'Some(Map(false -> NonEmptyList(-2, -5), true -> NonEmptyList(12, 3)))'
```
this seems to fail only on Scala 2.13.1 🤔 works for me locally and on the rest of Scala versions https://travis-ci.org/github/typelevel/cats/builds/691476028

Reply:
It works for you locally on 2.13?

Reply:
2.12.10

Reply:
quite not sure how to approach this - is there anything I can do to work it out or should I just stay with the explicit assertion?

Reply:
This is an issue with `toString` changing for some collection types between 2.12 and 2.13. Unfortunately I think you have to do something like this: #3155